src/components/Header.jsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { SearchBlock } from "./SearchBlock";
+import { Sorter } from "./Sorter";
+
+export class Header extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.onChangeQuery = this.onChangeQuery.bind(this);
+        this.onChangeSearchBy = this.onChangeSearchBy.bind(this);
+        this.onChangeSortBy = this.onChangeSortBy.bind(this);
+    }
+
+    onChangeSearchBy(name) {
+        this.props.onChangeSearchBy(name);
```

Comment:
you can leave as it is or you can avoid creating new functions and just call props instantly.
For instance: <SearchBlock
                        onChangeQuery={this.props.onChangeQuery}